modules/deprecations.nix

```
@@ -0,0 +1,40 @@
+{lib, ...}: let
+  inherit (builtins) match trace;
+  inherit (lib.asserts) assertMsg;
+  inherit (lib.modules) doRename;
+  inherit (lib.options) showOption;
+  inherit (lib.lists) concatLists;
+
+  mkRenamedOptionModuleUntil = from: to: date: let
+    isValidISODate = dateString: (match "[0-9]{4}-[0-9]{2}-[0-9]{2}" dateString != null);
+  in
+    assert assertMsg (isValidISODate date) "Cannot rename `${showOption from}` to `${showOption to}`: Your date needs to be in ISO 8601 format, ie. yyyy-mm-dd (got ${date} instead).";
```

The let in looks kinda funky, and this function might be used in another function if we add this for removed options. Just move it outside of it.

```suggestion

  isValidISODate = dateString: (match "[0-9]{4}-[0-9]{2}-[0-9]{2}" dateString != null);
    
  mkRenamedOptionModuleUntil = from: to: date:
    assert assertMsg (isValidISODate date) "Cannot rename `${showOption from}` to `${showOption to}`: Your date needs to be in ISO 8601 format, ie. yyyy-mm-dd (got ${date} instead).";
```